### PR TITLE
split topic endpoint parameter to endpoint and aggregation_type

### PR DESCRIPTION
**Pull request by Gigaszi**
### Description
Topic definitions now have one field for the used ohsome API endpoint and one field for the aggregation type.

### New or changed dependencies
- None

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

File: workers/tests/integrationtests/fixtures/vcr_cassettes/indicators/test_building_completeness.yaml [hunk 1]
```diff
@@ -70,4 +70,75 @@ interactions:
       - accept-encoding
     http_version: HTTP/1.1
     status_code: 200
+- request:
```
Comment (matthiasschaub) [inline workers/tests/integrationtests/fixtures/vcr_cassettes/indicators/test_building_completeness.yaml:73]: Why did this cassette change?

File: workers/ohsome_quality_analyst/topics/models.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ class TopicDefinition(BaseTopic):
     """Includes the ohsome API endpoint and parameters needed to retrieve the data."""
 
     endpoint: str
+    aggregation_type: str
```
Comment (matthiasschaub) [inline workers/ohsome_quality_analyst/topics/models.py:28]:
```suggestion
    aggregation_type: Literal["area", "count", "length", "perimeter"]

```

File: workers/tests/unittests/test_topic.py [hunk 3]
```diff
@@ -28,6 +28,7 @@ def test_topic_definition():
         indicators=["mapping-saturation"],
         project="core",
         endpoint="endpoint",
+        aggregation_type="aggregation_type",
```
Comment (matthiasschaub) [inline workers/tests/unittests/test_topic.py:31]: After using Literal in the topic model, this will fail.

**Review by matthiasschaub — CHANGES_REQUESTED** [review]

**Comment by matthiasschaub** [comment]
Do not forget to update `docs/topic.md`

File: workers/tests/integrationtests/fixtures/vcr_cassettes/indicators/test_building_completeness.yaml (hunk 1 above)
Comment (Gigaszi) [reply workers/tests/integrationtests/fixtures/vcr_cassettes/indicators/test_building_completeness.yaml:73]: Done

File: workers/ohsome_quality_analyst/topics/models.py (hunk 2 above)
Comment (Gigaszi) [reply workers/ohsome_quality_analyst/topics/models.py]: Done

File: workers/tests/unittests/test_topic.py (hunk 3 above)
Comment (Gigaszi) [reply workers/tests/unittests/test_topic.py]: Done

File: CHANGELOG.md [hunk 4]
```diff
@@ -84,6 +84,7 @@
 - refactor: move base classes to related modules ([#524])
 - refactor: move topic-indicator-combinations to topic definition ([#528])
 - tests(vcr): don't record local requests and change cassettes directory structure ([#579])
+- split 'endpoint' parameter of topics to `endpoint` and `aggregation_type` ([#600])
```
Comment (matthiasschaub) [inline CHANGELOG.md:87]: Add this to a new section `# current main`. A release happened since this PR.

**Review by matthiasschaub — APPROVED** [review]

**Pull request merged**